### PR TITLE
Settings Sync: Enum Cleanup

### PR DESCRIPTION
**Pull request by bjtitus**
| 📘 Part of: #1400 |
|:---:|

This fixes a number of mismatches with enum types with old and new versions, and some mismatches with the documented version from Android:

* `PodcastEpisodeSortOrder`
* `LibraryType` (Grid Layout) and `LibrarySort` (Grid Sort)
* `UploadedSort` (Files sort order)
* Bookmark sort for episodes and podcasts

I've also added unit tests for importing all enums in `PodcastSettings` and `AppSettings` which caught a few issues which were addressed by the enum case fixes.

## To test

> [!IMPORTANT]
> The `newSettingsStorage` and `settingsSync` flags must be enabled

### Grid Layout and Grid Sort
CleanShot 2024-03-19 at 14 14 01@2x
* D1: Change the Sort By and Layout types from the Podcasts grid/list view
* D1: Refresh the Podcasts list
* D2: Refresh the Podcasts list
* D2: Ensure that the sort order and layout match

### Episode sort

CleanShot 2024-03-19 at 14 15 31@2x

* D1: Navigate to a Podcast's page
* D1: Change the "Sort Episodes" sort order
* D1: Refresh Podcasts list
* D2: Refresh Podcasts list
* D2: Navigate to the same podcast
* D2: Verify that sort order matches

#### Episodes sort
* D1: Add some Bookmarks to an episode
* D1: Navigate to the Episode's Bookmarks page
* D1: Change the sort order by tapping the ⋯ button
* D1: Refresh the Podcasts list
* D2: Navigate to the same Episode's Bookmarks page
* D2: Verify the sort order

#### Podcast sort
* D1: Navigate to the Podcast's Bookmarks page
* D1: Change the sort order by tapping the ⋯ button
* D1: Refresh the Podcasts list
* D2: Navigate to the same Podcast's Bookmarks page
* D2: Verify the sort order

#### Files Sort Order

* D1: Navigate to Profile > Files and tap the ⋯
* D1: Change the "Sort By" ordering
* D1: Verify that sort order changes as expected
* D1: Refresh profile
* D2: Refresh profile
* D2: Check the Profile > Files section and make sure episode order and sort type match

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/43">7.60 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Pocket Casts Watch App/PodcastEpisodeListViewModel.swift [hunk 1]
```diff
@@ -12,6 +12,10 @@ class PodcastEpisodeListViewModel: ObservableObject {
         let sortStr: String
         let sortOrder = episodeSortOrder ?? PodcastEpisodeSortOrder.newestToOldest
         switch sortOrder {
+        case .titleAtoZ:
+            sortStr = "ORDER BY title DESC, addedDate"
+        case .titleZtoA:
+            sortStr = "ORDER BY title ASC, addedDate"
```
Comment (leandroalonso) [inline Pocket Casts Watch App/PodcastEpisodeListViewModel.swift]: These are inverted

```suggestion
        case .titleAtoZ:
            sortStr = "ORDER BY title ASC, addedDate"
        case .titleZtoA:
            sortStr = "ORDER BY title DESC, addedDate"
```

File: podcasts/EpisodesDataManager.swift [hunk 2]
```diff
@@ -83,6 +83,10 @@ class EpisodesDataManager {
 
         let sortOrder = episodeSortOrder ?? PodcastEpisodeSortOrder.newestToOldest
         switch sortOrder {
+        case .titleAtoZ:
+            sortStr = "ORDER BY title DESC, addedDate"
+        case .titleZtoA:
+            sortStr = "ORDER BY title ASC, addedDate"
```
Comment (leandroalonso) [inline podcasts/EpisodesDataManager.swift]: Also inverted

```suggestion
        case .titleAtoZ:
            sortStr = "ORDER BY title ASC, addedDate"
        case .titleZtoA:
            sortStr = "ORDER BY title DESC, addedDate"
```

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@bjtitus tested all the options and they are syncing correctly, it also stays correct after disabling the flags.

However, some sorting order is inverted for newly added sort options.

**Comment by bjtitus** [comment]
🤦 Sorry! All fixed up now 👍 

https://github.com/Automattic/pocket-casts-ios/assets/3250/d7818aad-a38d-4ed5-9cb5-82e49289f7ee

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**